src/DynamoCore/Search/SearchElements/ZeroTouchSearchElement.cs

```diff
@@ -16,16 +16,17 @@ namespace Dynamo.Search.SearchElements
     public class ZeroTouchSearchElement : NodeSearchElement
     {
         private readonly FunctionDescriptor functionDescriptor;
+        private readonly string fullname;
 
         /// <summary>
         /// The name that is used during node creation
         /// </summary>
         public override string CreationName { get { return functionDescriptor != null ? functionDescriptor.MangledName : this.Name; } }
 
         /// <summary>
-        ///     The full name of entry which consists of qualified name for function descriptor.
+        ///     The full name of entry which consists of assembly name and qualified name for function descriptor.
         /// </summary>
-        public override string FullName { get { return functionDescriptor != null ? functionDescriptor.QualifiedName : base.FullName; } }
+        public override string FullName { get { return fullname; } }
```

Comment:
The previous logic was weird because `this.functionDescriptor` will never be `null` (the only constructor makes sure that is the case, or else it would have crashed the constructor itself).